felix/bpf/counters/counters.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package counters
+
+import (
+	"encoding/binary"
+	"fmt"
+	"runtime"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+
+	"github.com/projectcalico/calico/felix/bpf"
+	"github.com/projectcalico/calico/felix/bpf/libbpf"
+	"github.com/projectcalico/calico/felix/bpf/tc"
+)
+
+const (
+	MaxCounterNumber int = 8
+	uint32Size       int = 4
+)
+
+const (
+	TotalPackets = iota
+	ErrShortPacket
+	ErrFailedCSUM
+	AcceptedByFailsafe
+	AcceptedByPolicy
+	DroppedByPolicy
+)
+
+const (
+	HookIngress = "ingress"
+	HookEgress  = "egress"
+)
+
+var Hooks = []string{HookIngress, HookEgress}
+
+type Counters struct {
+	maps     map[string]bpf.Map
```

Comment:
wouldn't an array of 2 be enough? You can define constants for 0 and 1 for the code to be equally readable

Reply:
We will need another one for `XDP` later. To answer you question, I thought it would be nice to use the key value i.e. `ingress`, and `egress` in the log output. Does it make sense?

Reply:
Fixed.